src/vm/mod.rs

```diff
@@ -1,3 +1,115 @@
 // SPDX-License-Identifier: Apache-2.0
 
 mod linux;
+
+use crate::vm::linux::{
+    ioctl::{Cmd, CmdId},
+    types::{Capabilities, CpuidConfig, NR_CPUID_CONFIGS},
+};
+
+/// Handle to the KVM VM file descriptor
+pub struct Vm(pub kvm_ioctls::VmFd);
+
+impl Vm {
+    /// Retrieve information about the Intel TDX module
+    ///
+    /// # Example:
+    /// ```
+    /// use kvm_ioctls::Kvm;
+    /// use tdx::vm::Vm;
+    ///
+    /// let kvm_fd = Kvm::new().unwrap();
+    /// let vm_fd = kvm_fd.create_vm_with_type(2).unwrap();
+    /// let caps = Vm(vm_fd).get_capabilities().unwrap();
+    ///
+    /// println!("{:#?}", caps);
+    /// ```
+    pub fn get_capabilities(&self) -> Result<TDXCapabilities, kvm_ioctls::Error> {
+        // configure the kernel facing capabilities structure
+        let cpuid_configs: [CpuidConfig; NR_CPUID_CONFIGS] = [Default::default(); NR_CPUID_CONFIGS];
+
+        let caps = Capabilities {
+            attrs_fixed0: 0,
+            attrs_fixed1: 0,
+            xfam_fixed0: 0,
+            xfam_fixed1: 0,
+            supported_gpaw: 0,
+            padding: 0,
+            reserved: [0; 251],
+
+            nr_cpuid_configs: NR_CPUID_CONFIGS as u32,
+            cpuid_configs,
+        };
```

Comment:
Let's define a method for `Capabilities` called `default_with_num_configs` that would do heap allocation for the CPUID configs and zero-out all other data. So we can do the following:
```suggestion
        let caps = Capabilities::default_with_num_configs(num_cpuid_config);
```